source/template.yaml

```diff
@@ -0,0 +1,161 @@
+AWSTemplateFormatVersion: '2010-09-09'
+Transform: AWS::Serverless-2016-10-31
+
+Parameters:
+  gitsha:
+    Type: String
+    Description: Git sha 
+  environment:
+    Type: String
+    Description: Deployment environment
+    Default: "dev"
+
+Metadata:
+  AWS::ServerlessRepo::Application:
+    Name: login-idp-functions
+    Description: login.gov idp functions
+    Author: logindotgov
+    SpdxLicenseId: CC0-1.0
+    HomePageUrl: https://github.com/18f/identity-idp-functions
+    SemanticVersion: 0.0.1
+    SourceCodeUrl: https://github.com/18f/identity-idp-functions
+  AWS::CloudFormation::Interface:
+    ParameterGroups:
+      -
+        Label:
+          default: Application Settings
+        Parameters:
+          - gitsha
+          - environment
+
+Description: >
+  identity-idp-functions
+
+  IDP functions for login.gov
+
+# More info about Globals: https://github.com/awslabs/serverless-application-model/blob/master/docs/globals.rst
+Globals:
+  Function:
+    Timeout: 60
+    Runtime: ruby2.7
+
+Resources:
+  HelloWorldFunction:
+    Type: AWS::Serverless::Function # More info about Function Resource: https://github.com/awslabs/serverless-application-model/blob/master/versions/2016-10-31.md#awsserverlessfunction
+    Properties:
+      CodeUri: demo-function/hello_world/
+      Handler: app.lambda_handler
+      MemorySize: 128
+      AutoPublishAlias: !Ref environment
+      DeploymentPreference:
+        Type: AllAtOnce
+      VersionDescription: !Ref gitsha
+      Layers:
+        - !Ref AWSRubySDKLayer
+      Environment:
+        Variables:
+          S3_BUCKET_NAME: !Sub 
+            - login-gov-idp-doc-capture-${Environment}.${AccountId}-${Region}
+            - Environment: !Ref environment 
+              AccountId: !Ref AWS::AccountId
+              Region: !Ref AWS::Region
+          KMS_KEY_ALIAS: !Sub
+            - ${Environment}-idp-doc-capture
+            - Environment: !Ref environment
+          ENVIRONMENT_NAME: !Ref environment
+      Policies:
+        - Statement:
+          - Sid: S3ObjectAccess
+            Effect: Allow
+            Action:
+            - s3:PutObject
+            - s3:GetObject
+            - s3:DeleteObject
+            Resource: !Sub
+              - arn:aws:s3:::login-gov-idp-doc-capture-${Environment}.${AccountId}-${Region}/*
```

Comment:
Where does this name come from? Our terraform configs?

Reply:
Environment is a parameter for the deployment.  AccountId and Region are pulled from the profile being used for deployment.